src/components/SchemaRow/TopLevelSchemaRow.tsx

```diff
@@ -1,85 +1,106 @@
-            matchTriggerWidth
-            items={choices.map((choice, index) => ({
-              id: index,
-              title: choice.title,
-              onPress: () => setSelectedChoice(choice),
-            }))}
-            renderTrigger={props => (
-              <Pressable {...props}>
-                <Flex>
-                  <Flex mr={2} fontFamily="mono" fontWeight="semibold" fontSize="base" cursor="pointer" py={2}>
-                    {selectedChoice.title}
-                    <Box ml={1}>
-                      <Icon icon={faCaretDown} />
-                    </Box>
-                  </Flex>
-                  {combiner !== null ? (
-                    <Flex alignItems="center" color="muted">
-                      {combiner}
-                    </Flex>
-                  ) : null}
+      <>
+        <ScrollCheck />
+
+        <Menu
+          aria-label="Pick a type"
+          closeOnPress
+          matchTriggerWidth
+          items={choices.map((choice, index) => ({
+            id: index,
+            title: choice.title,
+            onPress: () => setSelectedChoice(choice),
+          }))}
+          renderTrigger={props => (
+            <Pressable {...props}>
+              <Flex>
+                <Flex fontFamily="mono" fontWeight="semibold" fontSize="base" cursor="pointer" pb={4}>
+                  {selectedChoice.title}
+                  <Box ml={1}>
+                    <Icon icon={faCaretDown} />
+                  </Box>
                 </Flex>
-              </Pressable>
-            )}
-          />
 
-          {childNodes.length > 0 ? <ChildStack childNodes={childNodes} currentNestingLevel={nestingLevel} /> : null}
-        </Box>
-      </DecreaseIndentation>
+                {combiner !== null ? (
+                  <Flex alignItems="center" color="muted">
+                    {combiner}
+                  </Flex>
+                ) : null}
+              </Flex>
+            </Pressable>
+          )}
+        />
+
+        {childNodes.length > 0 ? <ChildStack childNodes={childNodes} currentNestingLevel={nestingLevel} /> : null}
+      </>
     );
   }
 
   if (isComplexArray(schemaNode) && isPureObjectNode(schemaNode.children[0])) {
     return (
-      <DecreaseIndentation>
-        <Box pos="relative">
-          <Box mr={2} fontFamily="mono" fontWeight="semibold" fontSize="base" py={2}>
-            array of:
-          </Box>
-          {childNodes.length > 0 ? <ChildStack childNodes={childNodes} currentNestingLevel={nestingLevel} /> : null}
+      <>
+        <ScrollCheck />
+
+        <Box fontFamily="mono" fontWeight="semibold" fontSize="base" pb={4}>
+          array of:
         </Box>
-      </DecreaseIndentation>
+
+        {childNodes.length > 0 ? <ChildStack childNodes={childNodes} currentNestingLevel={nestingLevel} /> : null}
+      </>
     );
   }
 
-  return <SchemaRow schemaNode={schemaNode} nestingLevel={nestingLevel} />;
+  return (
+    <>
+      <ScrollCheck />
```

Comment:
Unless something has changed, I recall `IntersectionObserver` a pretty bad impact on perf when you had thousands of instances.
This may be not the case here due to a smaller quantity of nodes, but it's certainly something to watch out for.
I'll take a closer look at it later today

Reply:
There's only one instance though?

Reply:
Per each row no?

Reply:
Just for the `TopLevelSchemaRow`, which is only rendered once at the top

Reply:
ah yeah, true, me blind again.